### PR TITLE
Block movement now locks turning, migrate old save files to Ctrl

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
The block movement key now correctly locks turning like it used to.

Old save files are now migrated to Ctrl if they didn't bind it to anything before. This correctly replicates the old behavior.

## Why It's Good For The Game
It feels like we just removed a feature for people who don't follow the codebase. This is weird.

## Changelog
:cl:
fix: The block movement key now locks turning like it used to.
/:cl:

### CODE REVIEW

File: code/modules/client/preferences_savefile.dm [hunk 1]
```diff
@@ -73,6 +73,19 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
...
+
+		look_through_bindings:
+			for (var/list/key in key_bindings)
+				for (var/bind in key)
+					if (bind == "block_movement")
+						found_block_movement = TRUE
+						break look_through_bindings
+
+		if (!found_block_movement)
+			LAZYADD(key_bindings["Ctrl"], "block_movement")
```
Comment (Rohesie) [inline code/modules/client/preferences_savefile.dm:87]:
```suggestion
			LAZYADD(key_bindings["Alt"], "block_movement")
```
Default is now Alt

File: code/modules/client/preferences_savefile.dm [hunk 2]
```diff
@@ -73,6 +73,19 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
...
 
+	if (current_version < 38)
+		var/found_block_movement = FALSE
+
+		look_through_bindings:
+			for (var/list/key in key_bindings)
+				for (var/bind in key)
+					if (bind == "block_movement")
+						found_block_movement = TRUE
+						break look_through_bindings
```
Comment (Rohesie) [inline code/modules/client/preferences_savefile.dm:84]:
```suggestion
		for (var/list/key in key_bindings)
			for (var/bind in key)
				if (bind == "block_movement")
					found_block_movement = TRUE
					break
			if(found_block_movement)
				break
```
goto very evil

File: code/modules/client/preferences_savefile.dm (hunk 1 above)
Comment (Jared-Fogle) [reply code/modules/client/preferences_savefile.dm:87]: This was intentional. *The default* is. What people have been used to for years isn't--this was intended to make players not feel like there was a regression.

File: code/modules/client/preferences_savefile.dm (hunk 2 above)
Comment (Jared-Fogle) [reply code/modules/client/preferences_savefile.dm:84]: I was told named breaks are okay, but I can live with this.

File: code/modules/client/preferences_savefile.dm (hunk 1 above)
Comment (Rohesie) [reply code/modules/client/preferences_savefile.dm:88]: Ah, that's fair.

File: code/modules/client/preferences_savefile.dm (hunk 2 above)
Comment (Rohesie) [reply code/modules/client/preferences_savefile.dm]: I suppose we don't have a hard ruling on this. I dislike it immensely. Unless you are running very hot code, there's absolutely no need for this, you can always replicate the behavior with a control var.

**Pull request merged**